src/connectors/answers/connectAnswers.ts

```diff
@@ -0,0 +1,179 @@
+import {
+  checkRendering,
+  createDocumentationMessageGenerator,
+  createConcurrentSafePromise,
+  debounce,
+  addAbsolutePosition,
+  noop,
+} from '../../lib/utils';
+import { Connector, Hits, Hit, FindAnswersResponse } from '../../types';
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'answers',
+  connector: true,
+});
+
+export type AnswersRendererOptions = {
+  /**
+   * The matched hits from Algolia API.
+   */
+  hits: Hits;
+
+  /**
+   * whether it's still loading the results from the /answers API
+   */
+  isLoading: boolean;
+};
+
+export type AnswersConnectorParams = {
+  /**
+   * Attributes to use for predictions.
+   * If empty, we use all `searchableAttributes` to find answers.
+   * All your `attributesForPrediction` must be part of your `searchableAttributes`.
+   */
+  attributesForPrediction?: string[];
+
+  /**
+   * The languages in the query. Currently only supports `en`.
+   */
+  queryLanguages: string[];
+
+  /**
+   * Maximum number of answers to retrieve from the Answers Engine.
+   * Cannot be greater than 1000.
+   */
+  nbHits?: number;
+};
+
+export type AnswersConnector = Connector<
+  AnswersRendererOptions,
+  AnswersConnectorParams
+>;
+
+const connectAnswers: AnswersConnector = function connectAnswers(
+  renderFn,
+  unmountFn = noop
+) {
+  checkRendering(renderFn, withUsage());
+
+  return widgetParams => {
+    const { queryLanguages, attributesForPrediction = ['*'], nbHits = 1 } =
+      widgetParams || ({} as typeof widgetParams);
+
+    if (!queryLanguages || queryLanguages.length === 0) {
+      throw new Error(
+        withUsage('The `queryLanguages` expects an array of strings.')
+      );
+    }
+
+    const runConcurrentSafePromise = createConcurrentSafePromise<
+      FindAnswersResponse<Hit>
+    >();
+
+    let lastAnswersResult: Partial<FindAnswersResponse<Hit>>;
+    let isLoading = false;
+    const debouncedRenderFn = debounce(renderFn, 200);
+
+    return {
+      $$type: 'ais.answers',
+
+      init(initOptions) {
+        const { instantSearchInstance } = initOptions;
+        // eslint-disable-next-line no-warning-comments
+        // FIXME: remove this customization once the engine accepts url encoded query params
+        if (instantSearchInstance.client.transporter) {
+          instantSearchInstance.client.transporter.userAgent.value =
+            'answers-test';
+        }
```

Comment:
I'm not sure, but I wonder if it should be user land instead?

Reply:
I'm not too sure, either, but if we guide people to do it user-land, that code might stick forever, whereas this can be removed in the later version when the engine fixes the bug.

Reply:
I wonder if the user agent can be overridden from the requestOptions, have you tried that? You will set the userAgent for the InstantSearch session like this as well

Reply:
Yeah it works! 
https://github.com/algolia/instantsearch.js/pull/4581/commits/94e607360a002f8a7c18a7259e79522c3acc95f5